internal/host/statemachine.go

```diff
@@ -549,6 +549,22 @@ func NewHostStateMachine(sm stateswitch.StateMachine, th TransitionHandler) stat
 		},
 	})
 
+	sm.AddTransitionRule(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeRefresh,
+		SourceStates: []stateswitch.State{
+			stateswitch.State(models.HostStatusPreparingForInstallation),
+			stateswitch.State(models.HostStatusPreparingFailed),
+			stateswitch.State(models.HostStatusKnown),
+		},
+		Condition:        stateswitch.And(If(IsConnected), If(IsMediaConnected), th.IsPreparingTimedOut, stateswitch.And(stateswitch.Or(installationDiskSpeedUnknown, imagesAvailabilityUnknown), allConditionsSuccessfulOrUnknown)),
+		DestinationState: stateswitch.State(models.HostStatusInsufficient),
```

Comment:
If destination state is `insufficient`, will the next refresh move it to `known`?

Reply:
The aim here is to raise an event, the further transition of the host should follow the same pattern as for failure so I don't see a problem here?

Reply:
The error cases for disk speed check and image availability eventually end up "insufficient" anyway, so the focus of this transition is mainly to make sure that an event is raised as appropriate. Whatever the host transitions to afterwards is not a massive concern.

Especially as the timeout mirrors the same state changes we use for the error state.

Reply:
The short answer to `If destination state is insufficient, will the next refresh move it to known` is "yes"

Reply:
Seems strange behavior, that first refresh will move to one state, and the second one moves to another.  Do we have such behavior in different place?

Reply:
Agreed in conversation that we will have this transition straight to "Known"